### PR TITLE
KOGITO-6771 GHA: Add Validate Formatting Step

### PR DESCRIPTION
**Pull request by evacchi**
https://github.com/kiegroup/drools/pull/4208
https://github.com/kiegroup/kogito-runtimes/pull/1997
https://github.com/kiegroup/kogito-pipelines/pull/383

### CODE REVIEW

File: .github/workflows/validate-formatting.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: Maven Build
+    steps:
+      - name: Java and Maven Setup
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java-version }}
+          maven-version: ${{ matrix.maven-version }}
+          cache-key-prefix: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}
```
Comment (radtriste) [inline .github/workflows/validate-formatting.yml:36]: I guess you should reuse https://github.com/kiegroup/kogito-pipelines/tree/main/.ci/actions/maven

See https://github.com/kiegroup/kogito-runtimes/blob/main/.github/workflows/kogito-runtimes-pr.yml#L36

```suggestion
      - name: Java and Maven Setup
        uses: kiegroup/kogito-pipelines/.ci/actions/maven@main
        with:
          java-version: ${{ matrix.java-version }}
          maven-version: ${{ matrix.maven-version }}
          cache-key-prefix: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}
```

File: .github/workflows/validate-formatting.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    name: Maven Build
+    steps:
+      - name: Java and Maven Setup
+        uses: kiegroup/kogito-pipelines/.ci/actions/maven@main
+        with:
+          java-version: ${{ matrix.java-version }}
+          maven-version: ${{ matrix.maven-version }}
+          cache-key-prefix: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}
```
Comment (radtriste) [inline .github/workflows/validate-formatting.yml:36]:
```suggestion
      - name: Checkout current repository on main
        uses: actions/checkout@v2
      - name: Java and Maven Setup
        uses: kiegroup/kogito-pipelines/.ci/actions/maven@main
        with:
          java-version: ${{ matrix.java-version }}
          maven-version: ${{ matrix.maven-version }}
          cache-key-prefix: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}
```

**Comment by radtriste** [comment]
Validate formatting issue due to Nexus ...

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#1140` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 431
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1140/testReport/org.kie.kogito.examples/CallbackRestIT/Build_projects___testCallbackRest/">Build projects / org.kie.kogito.examples.CallbackRestIT.testCallbackRest</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.examples.CallbackRestIT that uses java.lang.String 1 expectation failed.<br/>Expected status code <404> but was <200>.<br/> within 30 seconds.<br/>	at org.kie.kogito.examples.CallbackRestIT.testCallbackRest(CallbackRestIT.java:50)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <404> but was <200>.<br/><br/>	at org.kie.kogito.examples.CallbackRestIT.lambda$testCallbackRest$0(CallbackRestIT.java:55)<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1140/testReport/org.acme.travel.tests.multimessaging.springboot/MultiMessagingIT/testProcess/">org.acme.travel.tests.multimessaging.springboot.MultiMessagingIT.testProcess</a></summary>
org.opentest4j.AssertionFailedError: expected: <0> but was: <3><br/>	at org.acme.travel.tests.multimessaging.springboot.MultiMessagingIT.testProcess(MultiMessagingIT.java:96)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1140/display/redirect

**Review by mariofusco — APPROVED** [review]

**Pull request closed without merging**